Login Credentials and Role Handling Feature

kerrlabajo opened the pull request:
This PR will be responsible for the following implementations:
1. Creating a login form.
2. Role retrieval for admin/employee.
3. Admin can build and push training image to ECR.
4. Admin can use this ECR URI for training.
5. Employees must enter their admin's ECR URI before using the app.

In order to open the task list in your Visual Studio, click `Ctrl+\+T` or search `Task List`.
Make sure to remove the task that you have implemented/completed.

kerrlabajo committed:
Add TODO tasks and remove old tasks

This PR will be responsible for the following implementations:
1. Creating a login form.
2. Role retrieval for admin/employee.
3. Admin can build and push training image to ECR.
4. Admin can use this ECR URI for training.
5. Employees must enter their admin's ECR URI before using the app.

kerrlabajo committed:
Merge branch `development` that had previous merge from PR #21

Ra-Jay committed:
Add ne form for creating aws connection

Ra-Jay committed:
add create connection form

Ra-Jay committed:
add method to verify the user role

Ra-Jay committed:
create image builder form

Ra-Jay committed:
Update create connection form

Ra-Jay committed:
add build image button for admin

Ra-Jay committed:
Update method to get the role details of user

Ra-Jay committed:
create a method to execute shellscript

Ra-Jay committed:
set ecr_uri and update role checking method

Ra-Jay committed:
update ExecuteShellScript to use powershell in executing the script

Ra-Jay committed:
Create a form for ecr input for employees

Ra-Jay committed:
Create a singleton class for storing user creds

Ra-Jay committed:
Save user credentials

Ra-Jay committed:
Handle user creds input

Ra-Jay committed:
handle closing the connection

Ra-Jay commented:
Updates:

- Create the following forms:
     - CreateConnectionForm
     - ImageBuilderForm
     - EmployeeEcrUriInputForm
- Create a singleton class for temporarily storing user credentials
- Event handler that saves the user input in creating connection
- Add a method to verify the user role using roleArn - identify if user is admin or employee
     - Enable Image builder if admin; otherwise, otherwise open EmployeeEcrUriInputForm for employees to input ecr uri directly
- Create a method that will execute the shell script on button click in ImageBuilder form
- Handle closing the connection

Initial app run: 
![image](https://github.com/kerrlabajo/LSC-Trainer/assets/58256966/bca55bd1-0f79-46b3-8700-a8061cf97cb5)
 
Form in creating connection: 
![image](https://github.com/kerrlabajo/LSC-Trainer/assets/58256966/39407d06-f145-45a6-a85f-496a34137932)

Form in building image:
![image](https://github.com/kerrlabajo/LSC-Trainer/assets/58256966/4ec5303a-5277-413a-9de0-2b18622e6cd5)

kerrlabajo committed:
Merge branch `features` from PR #20

kerrlabajo committed:
Resolve major outdated UI due to many changes from PR #20

kerrlabajo committed:
Remove unused event listeners

kerrlabajo committed:
Remove TODO comments

kerrlabajo committed:
Remove contextMenuStrip1 metadata

kerrlabajo committed:
Remove unused `_iamClient`

kerrlabajo committed:
Rename to `helpMenu` and `newTrainingJobMenu_Click` instead

kerrlabajo committed:
Remove extra components

kerrlabajo committed:
Rename proper btn naming on click behavior methods

kerrlabajo committed:
Remove `OtherForm_FormClosed`

kerrlabajo committed:
Set name to `testConnectionMenu_Click`

kerrlabajo committed:
Comment out `IsAdminRole`

kerrlabajo committed:
Set proper label and textbox naming as well as organize tab indeces

kerrlabajo committed:
Remove `comboBox1_SelectedValueChanged` behavior

kerrlabajo committed:
Refactor to `btnConect_Click`

kerrlabajo committed:
Use `RegionCode` method and use items as the exact region

kerrlabajo committed:
Add title `Create Connection` for the `CreateConnectionForm`

kerrlabajo committed:
Set `CreateConnectionForm` as initial Form on startup but only when production

kerrlabajo committed:
Add `development` as an argument and private variable for `MainForm` and `CreateConnectionForm`

kerrlabajo committed:
Fix `closeConnectionMenu_Click`

kerrlabajo committed:
Arrange valid connection and development variable assigning

kerrlabajo committed:
Rearrange test and close connection positions

kerrlabajo committed:
Rename to `testConnnectionMenu_Click`

kerrlabajo committed:
Add `TestSageMakerClient` method

kerrlabajo committed:
Refactor to use `AWS_Herlper.TestSageMakerClient` instead

kerrlabajo committed:
Add `btnTestConnection` within `CreateConnectionForm`

kerrlabajo committed:
Add `btnTestConnection_Click` that has similar behavior to `testConnectionMenu`

kerrlabajo committed:
Add lacking using imports

kerrlabajo committed:
Add `regionCode` variable

kerrlabajo committed:
Remove `regionCode` variable instead

kerrlabajo commented:
Major update to the `Connection` Implementation:
1. The `CreateConnectionForm` will be first initialized before the `MainForm` and will only be opened later after testing the connection of the user inputs. This will only be applicable when `development` is false, in short, only upon production.
2. A `Test Connection` must be held first upon attempting to create connection before proceeding to the `MainForm`.
3. Generalized the `TestSageMakerClient` for said functionality for two different forms.
4. Clicking on `Close Connection` will close the `MainForm` and opens the `CreateConnectionForm` again.
5. Specific regions of interest are shown as items for the `regionDropdown` instead of its region code. These are then handled by `GetRegionCode` function.

In this update, there are still no ARN role extraction and "Admin" handler. To be implemented soon.

kerrlabajo committed:
Remove redundant `btnRefresh`

kerrlabajo committed:
Merge branch 'features' into feat/login-credentials

kerrlabajo committed:
Use `buildImageMenu` instead of `btnBuildImage`

kerrlabajo committed:
Add `CheckAdministratorRole` implementation

kerrlabajo committed:
Modify `ImageBuilderForm` design

Modifications:
1. Remove redundant account id and region. This is because it can reuse the previous details submitted from create connection.
2. Rearrange layout, size, and position of UI.
3. Rename label and set tab indeces orderly.

kerrlabajo committed:
Add `ACCOUNT_ID`

kerrlabajo committed:
Add new private variables for `ImageBuilderForm`

kerrlabajo committed:
Remove redundant variables

kerrlabajo committed:
Install `AWSSDK.EC2` nuget package

kerrlabajo committed:
Implement `LaunchInstancePushPrivate` method

kerrlabajo committed:
Merge branch 'features' into feat/login-credentials

kerrlabajo committed:
Readd deleted `cloudWatchLogsClient` assignment

kerrlabajo committed:
Merge branch 'features' into feat/login-credentials

kerrlabajo committed:
Fetch parent directory only and remove duplicates

kerrlabajo committed:
Set `outputKey` and `modelKey` once `trainingJobOutputs` are fetched

kerrlabajo committed:
Rename to `btnFetchOutput` and `outputListComboBox`

kerrlabajo committed:
Rename to `GetTrainingJobOutputList`

kerrlabajo committed:
Reflect name changes for `btnFetchOutput` and `outputListComboBox`

kerrlabajo committed:
Restructure `trainingJobName`

kerrlabajo committed:
Refactor to `btnFetchOutput_Click`

kerrlabajo committed:
Set cursor wait and panel enablers upon clicking upload to s3 and fetch output

kerrlabajo committed:
Merge branch 'features' into feat/login-credentials

kerrlabajo committed:
Rename labels and button

kerrlabajo committed:
Refactor to `btnContinue_Click`

kerrlabajo committed:
Fix inconsistent button enablers and cursor waiters

kerrlabajo committed:
Add `mainForm` variable in order to access `IsImageBuilt`

kerrlabajo committed:
Add `mainForm` as parameter in `ImageBuilderForm` creation

kerrlabajo committed:
Make `MainForm.development` public

kerrlabajo committed:
Add `mainForm` var in `CreateConnectionForm` and add as paramter when creation

kerrlabajo committed:
Handle upon the mock successful `buildButton_Click`

kerrlabajo committed:
Reflect adding `MainForm` creation as parameter for `CreateConnectionForm`

kerrlabajo committed:
Add logger for user and dev details

kerrlabajo committed:
Add `isAdminRole`

kerrlabajo committed:
Add `mainForm` var in EmployeEcrUriInputForm` and handle return to `mainForm`

kerrlabajo committed:
Implement role handling

Modifications:
1. Show `buildImageMenu` if `isAdminRole`.
2. Always set `ECR_URI` according to `UserConnectionInfo.EcrUri` only except when `development` is true.
3. Handle `Administrator` role that have no image built yet.
4. Handle `Administrator` role with a built image. (Incomplete build and push image. And will keep on asking `Administrator` to build image first before starting even though they have pushed. Will fix using user config stored locally).
5. Handle `Employee` role who cannot see `buildImageMenu` and only have to fill up the ECR URI details.
6. Add logger for all details confirmed.

kerrlabajo committed:
Add placeholder value for `ROLE_ARN`

kerrlabajo committed:
Add comment in program
